builtin/providers/aws/resource_aws_autoscaling_scaling_policy.go

```diff
@@ -0,0 +1,182 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/service/autoscaling"
+)
+
+func resourceAwsAutoscalingScalingPolicy() *schema.Resource {
+    return &schema.Resource{
+        Create: resourceAwsAutoscalingScalingPolicyCreate,
+        Read:   resourceAwsAutoscalingScalingPolicyRead,
+        Update: resourceAwsAutoscalingScalingPolicyUpdate,
+        Delete: resourceAwsAutoscalingScalingPolicyDelete,
+
+        Schema: map[string]*schema.Schema{
+            "adjustment_type": &schema.Schema{
+                Type: schema.TypeString,
+                Required: true,
+            },
+            "autoscaling_group_name": &schema.Schema{
+                Type: schema.TypeString,
+                Required: true,
+                ForceNew: true,
+            },
+            "cooldown": &schema.Schema{
+                Type: schema.TypeInt,
+                Optional: true,
+            },
+            "min_adjustment_step": &schema.Schema{
+                Type: schema.TypeInt,
+                Optional: true,
+            },
+            "policy_arn": &schema.Schema{
+                Type: schema.TypeString,
+                Computed: true,
+            },
+            "policy_name": &schema.Schema{
+                Type: schema.TypeString,
+                Required: true,
+                ForceNew: true,
+            },
+            "scaling_adjustment": &schema.Schema{
+                Type: schema.TypeInt,
+                Required: true,
+            },
+        },
+    }
+}
+
+func resourceAwsAutoscalingScalingPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+    autoscalingconn := meta.(*AWSClient).autoscalingconn
+
+    params := getAwsAutoscalingPutScalingPolicyInput(d)
+
+    log.Printf("[DEBUG] AutoScaling PutScalingPolicy: %#v", params)
+    resp, err := autoscalingconn.PutScalingPolicy(&params)
+    if err != nil {
+        return fmt.Errorf("Error putting scaling policy: %s", err)
+    }
+
+    d.Set("policy_arn", resp.PolicyARN)
+    d.SetId(d.Get("policy_name").(string))
+    log.Printf("[INFO] AutoScaling Scaling PolicyARN: %s", d.Get("policy_arn").(string))
+
+    return resourceAwsAutoscalingScalingPolicyRead(d, meta)
+}
+
+func resourceAwsAutoscalingScalingPolicyRead(d *schema.ResourceData, meta interface{}) error {
+    p, err := getAwsAutoscalingScalingPolicy(d, meta)
+    if err != nil {
+        return err
+    }
+    if p == nil {
```

Comment:
if `p` isn't found, we should unset the resource id in the state file. I see you do that in the  `getAwsAutoscalingScalingPolicy` method, however I think it may make more sense to do it here. That way a method named `get*` doesn't possibly _remove_ something from your state file as a side affect.